src/legacy/ui/public/courier/fetch/components/shard_failure_modal.tsx

```diff
@@ -0,0 +1,136 @@
+
+export interface Props {
+  onClose: () => void;
+  request: Request;
+  response: ResponseWithShardFailure;
+  title: string;
+}
+
+export function ShardFailureModal({ request, response, title, onClose }: Props) {
+  if (!response || !response._shards || !Array.isArray(response._shards.failures) || !request) {
+    // this should never ever happen, but just in case
+    const errorTitle = i18n.translate(
+      'common.ui.courier.fetch.shardsFailedModal.componentErrorTitle',
+      {
+        defaultMessage: 'Sorry, there was an error',
+      }
+    );
+    const errorDesc = i18n.translate(
+      'common.ui.courier.fetch.shardsFailedModal.componentErrorDesc',
+      {
+        defaultMessage: 'The ShardFailureModal component received invalid properties',
+      }
+    );
+    return (
+      <EuiCallOut title={errorTitle} color="danger" iconType="alert">
+        {errorDesc}
+      </EuiCallOut>
+    );
+  }
+
+  const requestJSON = JSON.stringify(request, null, 2);
+  const responseJSON = JSON.stringify(response, null, 2);
+  const failures = response._shards.failures;
+
+  const tabs = [
+    {
+      id: 'table',
+      name: i18n.translate('common.ui.courier.fetch.shardsFailedModal.tabHeaderShardFailures', {
+        defaultMessage: 'Shard failures',
+        description: 'Name of the tab displaying shard failures',
+      }),
+      content: <ShardFailureTable failures={failures} />,
+    },
+    {
+      id: 'json-request',
+      name: i18n.translate('common.ui.courier.fetch.shardsFailedModal.tabHeaderRequest', {
+        defaultMessage: 'Request',
+        description: 'Name of the tab displaying the JSON request',
+      }),
+      content: (
+        <EuiCodeBlock language="json" isCopyable className="shardFailureModal__codeblock">
+          {requestJSON}
+        </EuiCodeBlock>
+      ),
+    },
+    {
+      id: 'json-response',
+      name: i18n.translate('common.ui.courier.fetch.shardsFailedModal.tabHeaderResponse', {
+        defaultMessage: 'Response',
+        description: 'Name of the tab displaying the JSON response',
+      }),
+      content: (
+        <EuiCodeBlock language="json" isCopyable className="shardFailureModal__codeblock">
+          {responseJSON}
+        </EuiCodeBlock>
+      ),
+    },
+  ];
+
+  return (
+    <React.Fragment>
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>{String(title)}</EuiModalHeaderTitle>
+      </EuiModalHeader>
+      <EuiModalBody>
+        <EuiTabbedContent tabs={tabs} initialSelectedTab={tabs[0]} autoFocus="selected" />
+      </EuiModalBody>
+      <EuiModalFooter>
+        <EuiCopy textToCopy={responseJSON}>
+          {copy => (
+            <EuiButtonEmpty onClick={copy}>
+              <FormattedMessage
+                id="common.ui.courier.fetch.shardsFailedModal.copyToClipboard"
+                defaultMessage="Copy response to clipboard"
+                description="Copy the response JSON  to clipboard"
```

Comment:
I don't think a text like that would need a `description`. I think it's rather self explanatory to the translator what it means, or/and the description itself doesn't make it much clearer for a translator. So I think we should just leave the `description` here. If we get feedback and questions from translators for that string, we can still add a detailed explanation later.

If you disagree and want to keep it: There is one redundant space between 'JSON' and 'to' :)

Reply:
no, I'm in favor of removing code if it's not necessary 👍